extensions/azurePublish/src/components/azureProvisionDialog.tsx

```diff
@@ -211,6 +221,28 @@ const reviewCols: IColumn[] = [
   },
 ];
 
+const getHostname = (config) => {
+  if (config?.hostname) {
+    return config.hostname;
+  } else if (config?.name) {
+    return config?.environment ? `${config.name}-${config.environment}` : config.name;
+  }
+};
+
+const getDefaultFormData = (currentProfile, defaults) => {
+  return {
+    choice: defaults.choice ?? 'create',
+    tenantId: currentProfile?.tenantId ?? defaults.tenantId,
```

Comment:
This is probably overthinking this, but I noticed that you clear the other fields when the tenantId changes.  Here, this chooses between the profile tenant and the default.  If we choose one or the other, would we want to use all the tenant-bound values from that choice?  (i.e. I choose the profile tenantId and so I want to choose the subscriptionId from the profile and not the default?)

Reply:
I think this could actually just be `tenantId: currentProfile?.tenantId` because we won't ever set the tenant id with the defaults.